src/OpenColorIO/ops/lut3d/Lut3DOpCPU.cpp

```diff
@@ -384,405 +417,412 @@ void Lut3DTetrahedralRenderer::apply(const void * inImg, void * outImg, long num
     const float * in = (const float *)inImg;
     float * out = (float *)outImg;
 
-#ifdef USE_SSE
-
-    __m128 step = _mm_set1_ps(m_step);
-    __m128 maxIdx = _mm_set1_ps((float)(m_dim - 1));
-    __m128i dim = _mm_set1_epi32(m_dim);
-
-    __m128 v[4];
-    OCIO_ALIGN(float cmpDelta[4]);
-
-    for (long i = 0; i < numPixels; ++i)
+    if (m_applyLutFunc)
     {
-        float newAlpha = (float)in[3];
-
-        __m128 data = _mm_set_ps(in[3], in[2], in[1], in[0]);
+        m_applyLutFunc(m_optLut, m_dim, in, out, numPixels);
+    }
+    else
+    {
+  #ifdef USE_SSE
```

Comment:
Questions:

- Do we still need the SSE code here? Isn't it replaced by applyTetrahedralSSE2 ?
- If we still need it, what are the scenarios?

Reply:
I think I only left there to help measure the performance difference, the code can be removed.